crates/wasi-common/winx/src/file.rs

```diff
@@ -415,3 +388,103 @@ pub fn change_time(file: &File) -> io::Result<i64> {
 
     Ok(tm)
 }
+
+pub fn query_access_information(handle: RawHandle) -> Result<AccessMode> {
+    let mut io_status_block = IO_STATUS_BLOCK::default();
+    let mut info = FILE_ACCESS_INFORMATION::default();
+
+    unsafe {
+        let status = NtQueryInformationFile(
```

Comment:
Is this syscall available on all platforms beginning with Windows 7? Or is it a more recent addition perhaps?

Reply:
It's an ancient NT API, going all the way back to Windows 2000.

Reply:
Ah, cool! Would you mind dropping a comment in the code here with the link to the relevant description in the MSDN pages for posteriority please?

Reply:
I'll add a comment and I'll also add some Windows-only unit tests to make sure these two new functions are operating correctly.

Reply:
Awesome, thanks!

Reply:
It appears that winapi doesn't currently support *any* ntdll functions: https://github.com/retep998/winapi-rs/commit/aaf9c29071860aa9eddb6fa08096556517ddc6a2

Reply:
I realized I didn't add the comment or the tests I said I would.  Unresolving this comment.